### PR TITLE
Implement jnp.unique with axis and size set.

### LINKED ISSUE #7090: Support size and axis argument for `jnp.unique`
Please:

- [x] Check for duplicate requests.
- [x] Describe your goal, and if possible provide a code snippet with a motivating example.

Hi,
at the moment it is not possible to have `jnp.unique` functioning in jit with `axis` being something else than `None`. While the `size` argument is supposed to make `jnp.unique` jit compatible, it doesn't work with also the `axis` argument being set.

A simple use case for this would be to find the unique vectors in a set.

### PR DESCRIPTION
**Pull request by n-gao**
Fixes #7090

Part of #3614

### CODE REVIEW

**Comment by jakevdp** [comment]
Thanks for taking a look at this! Once the implementation is finished, we'll also need to add tests for this new behavior; you can probably model them off of the existing tests (including the one added in #6912)

Please let me know if you run into any difficulties

**Comment by jakevdp** [comment]
Hi - I'd love to get this feature into JAX. Please let me know if there's any way I can help!

**Comment by jakevdp** [comment]
Hi - I'm going to close this PR for now, since it looks like work on it has stalled. Feel free to re-open if you want to continue work on this feature!
